technical_docs/Claims.md

```diff
@@ -0,0 +1,82 @@
+
+* *nbf* (Not Before): The interpretation for `nbf` is unchanged from the RFC, but this is considered a CRITICAL attribute for a SciToken.
+
+* *exp* (Expiration Time): The interpretation for `exp` is unchanged from the RFC, but this is considered a CRITICAL attribute for a SciToken.  The motivation behind the combination of `nbf` and `exp` being marked as critical is to allow resources to enforce maximum token lifetime limits.
+
+* *iss* (Issuer): The issuer of the SciTokens.  VOs are strongly encouraged to populate the field; it MAY contain information local to the VO context (such as the host issuing the token).  SciTokens uses the `vo` claim name to identify the issuer in the global context.
+
+SciToken-specific Claims
+------------------------
+
+The SciTokens project defines JWT claims specific to our problem domain.  From RFC 7519:
+
+>   A producer and consumer of a JWT MAY agree to use Claim Names that
+>   are Private Names: names that are not Registered Claim Names
+>   (Section 4.1) or Public Claim Names (Section 4.2).  Unlike Public
+>   Claim Names, Private Claim Names are subject to collision and should
+>   be used with caution.
+
+In the long-term, we hope to have our claim names registered with IANA.  Until then, we will
+utilize the URI form.  A SciTokens validator MUST accept either URI or non-URI form.
+
+* *vo* (Virtual Organization): https://scitokens.org/v1/vo. The virtual organization issuing the token.  This MUST be globally unique for the community utilizing the SciToken.  It MUST be utilized to determine the appropriate trust roots for token verification.  If present in a token, it MUST also be present in the header.  A token chain MUST have the VO specified in the base token.
+
+* *op* (Authorized Operation): https://scitokens.org/v1/op.  A list of operations the bearer is allowed to perform.  Known operations are:
+
+   * `read`: Read data.
+   * `write`: Write data.
+   * `queue`: Submit a task or a job to a queueing service.
+   * `execute`: Immediately launch or execute a task.
+
+   The operation definitions are currently kept open-ended and intended to be interpreted by the specific user community.  The `op` claim is OPTIONAL.
+
+* *svc* (Service): https://scitokens.org/v1/svc.  A service URI the SciToken is authorized to access.  For example, if the VO has write access to several storage services, this claim may be utilized to limit a token to a single endpoint.  The `svc` claim is OPTIONAL.
```

Comment:
I think this is the JWT [aud](https://tools.ietf.org/html/rfc7519#section-4.1.3) claim?